lib/bosh/BoshDirectorClient.js

```diff
@@ -45,19 +151,28 @@ class BoshDirectorClient extends HttpClient {
 
   /* Deployment operations */
 
-  getDeployments() {
-    return this
-      .request({
-        method: 'GET',
-        url: '/deployments'
-      }, 200)
-      .then(res => JSON.parse(res.body));
+  getDeployments(queryInactive) {
+    let configs = queryInactive === undefined ? this.activeConfigs : this.inactiveConfigs;
+    return Promise
+      .map(configs, directorConfig => {
+        return this
+          .makeRequestWithConfig({
+            method: 'GET',
+            url: '/deployments'
+          }, 200, directorConfig)
+          .then(res => JSON.parse(res.body))
+          .tap(deployments => _
+            .map(deployments, deployment => this
+              .cache[deployment.name] = directorConfig));
+      })
+      .reduce((all_deployments, deployments) => all_deployments.concat(deployments), []);
   }
 
   getDeploymentNames(queued) {
     const activeDeploymentNames = this
       .getDeployments()
-      .then(deployments => _.map(deployments, deployment => deployment.name));
+      .then(deployments =>
+        _.map(deployments, deployment => deployment.name));
     const queuedDeploymentNames = !queued ? [] : this
       .getTasks({
```

Comment:
getTasks need not go over all primary, it just needs to go over only the activePrimary configs.